oml/inference/base.py

```diff
@@ -0,0 +1,45 @@
+from pathlib import Path
+from typing import List, Optional
+
+import torch
+from torch import Tensor, nn
+from torch.utils.data import DataLoader
+from tqdm.auto import tqdm
+
+from oml.datasets.list_dataset import ListDataset
+from oml.transforms.images.utils import TTransforms, get_im_reader_for_transforms
+from oml.utils.images.images import TImReader
+
+
+# todo: use lightning for half precision and DDP
+def inference_on_images(
+    model: nn.Module,
+    paths: List[Path],
+    transform: TTransforms,
+    num_workers: int,
+    batch_size: int,
+    verbose: bool,
+    f_imread: Optional[TImReader] = None,
+) -> Tensor:
+    if f_imread is None:
+        f_imread = get_im_reader_for_transforms(transform)
+
+    prev_mode = model.training
+    model.eval()
+
+    dataset = ListDataset(paths, bboxes=None, transform=transform, f_imread=f_imread)
+    loader = DataLoader(dataset, batch_size=batch_size, num_workers=num_workers, shuffle=False)
+    device = next(model.parameters()).device
```

Comment:
The same thoughts here about introducing the `get_device` function, and the context manager to restore model training flag.

Reply:
done